include/powerloader/curl.hpp

```diff
@@ -35,17 +36,25 @@ namespace powerloader
         bool m_serious;
     };
 
-    struct POWERLOADER_API Response
+    struct POWERLOADER_API BaseResponse
     {
-        std::map<std::string, std::string> header;
-        mutable std::stringstream content;
+        std::map<std::string, std::string> headers;
 
-        curl_off_t avg_speed;
+        curl_off_t average_speed;
         curl_off_t downloaded_size;
         long http_status;
         std::string effective_url;
 
         bool ok() const;
+
+        tl::expected<std::string, std::out_of_range> get_header(const std::string& header) const;
+
+        void fill_values(CURLHandle& handle);
+    };
+
+    struct POWERLOADER_API Response : public BaseResponse
+    {
+        mutable std::stringstream content;
```

Comment:
Shouldnt this be a `string`? If I understand it's usage correctly, once written in it does not need to be re-written, so we can just store the end result. It would also allow to remove `mutable`.